volumes.cpp

```diff
@@ -1301,18 +1321,156 @@ SurfaceVolume& makeSurface( const SurfaceCard* card, VolumeCache* v){
     else{
       throw std::runtime_error( mnemonic + " is not a supported surface" );
     }
-    
-    if( card->getTransform().hasData() ){
-      const Transform& transform = card->getTransform().getData();
-      surface->setTransform( &transform );
-    }
-    
-    cache.insert( card, surface );
-    return *surface;
-    
+  } 
+  if( card->getTransform().hasData() ){
+    const Transform& transform = card->getTransform().getData();
+    surface->setTransform( &transform );
   }
   
+  cache.insert( card, surface );
+  return *surface;
+    
+}
   
+
+SurfaceVolume* FacetSurface( const std::string mnemonic, const std::vector< double > args, int facet ){
+  if( mnemonic == "rcc" ){
+    return rccFacet( args, facet );
+  }
+  else if(mnemonic == "box"){
+    return boxFacet( args, facet );
+  }
+  else if( mnemonic == "rpp"){
+    return rppFacet( args, facet );
+  }
+  else if( mnemonic == "hex" || mnemonic == "rhp" ){
+    return hexFacet( args, facet );
+  }
+  else if( mnemonic == "rec" ){
+    return recFacet( args, facet );
+  }
+  throw std::runtime_error( mnemonic + " does not have macrobody facet support at this time." );
+}    
+
+
+//This function is used to find D in the planar equation Ax + By + Cz - D = 0 
+//when given two vectors.
+//The positioning vector always uses the first three arguments; i determines 
+//the first argument of the vector to which the plane will be parallel.  If 
+//end is true, the plane is at the end of the second vector in relation to the 
+//end of the first vector, otherwise it intersects the end of the first vector.
+double planePoint( std::vector< double > args, int i, bool end ){
```

Comment:
Also also... perhaps overload the constructor for `PlaneSurface`, adding a version that takes two `Vector3d` arguments instead of a `Vector3d` and an offset.  Then you can move this function `planePoint` inside that function and call the other constructor. (although technically you'd have to move the bulk of that other constructor to a private function and call it from both)